### PR TITLE
C sharp exception

### PR DESCRIPTION
**Pull request by 9inebar**
homework for  c_sharp_exception lecture

### CODE REVIEW

File: CSharpException/Triangle.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public double Side3
+    {
+        get { return side3;}
+    }
+
+    public Triangle(double side1, double side2, double side3)
+    {
+        if (side1 != 0 && side2 != 0 && side3 != 0)
+        {
+            if (side1 < (side2 + side3) && side2 < (side3 + side1) && side3 < (side1 + side2))
```
Comment (MariaEroshevich) [inline CSharpException/Triangle.cs]: I think should be || instead of && in if constructor .

File: CSharpException/Triangle.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        get { return side2;}
+    }
+    public double Side3
+    {
+        get { return side3;}
+    }
+
+    public Triangle(double side1, double side2, double side3)
+    {
+        if (side1 != 0 && side2 != 0 && side3 != 0)
```
Comment (MariaEroshevich) [inline CSharpException/Triangle.cs]: move verification for sides to separate methods.

File: CSharpException/Program.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using CSharpException;
+
+
+Console.WriteLine("Please enter side 1 length");
+try {
+    double side1 = Convert.ToDouble(Console.ReadLine());
+    if (side1 == 0)
```
Comment (MariaEroshevich) [inline CSharpException/Program.cs]: you have already thrown the exception in Triangle class. So no need to throw in try construction. Here you need to enter numbers into console and pass them to object of Triangle class.

File: CSharpException/Triangle.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+namespace CSharpException;
+
+public class Triangle
+{
+    private double side1;
+    private double side2;
+    private double side3;
+
+    public double Side1
```
Comment (MariaEroshevich) [inline CSharpException/Triangle.cs:13]: add spaces between properties.

**Review by MariaEroshevich — CHANGES_REQUESTED** [review]

File: CSharpException/Program.cs (hunk 3 above)
Comment (9inebar) [reply CSharpException/Program.cs]: it's made for not entering 3 wrong numbers one by one but exit the program after any wrong value

Comment (MariaEroshevich) [reply CSharpException/Program.cs]: You can make up another way to check this case. Introduce a new bool variable(isNumberNull) and put try catch construct in a loop (while(!isNumberNull)). And if the user enter null makes this variable true and add "continue;" in if construction to return to condition in while loop to enter number again.

File: CSharpException/Triangle.cs (hunk 4 above)
Comment (MariaEroshevich) [reply CSharpException/Triangle.cs:13]: I haven't seen that you add spaces between properties.

File: CSharpException/Triangle.cs (hunk 1 above)
Comment (MariaEroshevich) [reply CSharpException/Triangle.cs]: the same for the second if.

**Review by MariaEroshevich — DISMISSED** [review]

File: CSharpException/Triangle.cs (hunk 1 above)
Comment (dnnyhlwdplc) [reply CSharpException/Triangle.cs]: Same comment :)

**Review by dnnyhlwdplc — CHANGES_REQUESTED** [review]

File: CSharpException/Program.cs (hunk 3 above)
Comment (dnnyhlwdplc) [reply CSharpException/Program.cs]: same comment :)

File: CSharpException/Triangle.cs (hunk 2 above)
Comment (dnnyhlwdplc) [reply CSharpException/Triangle.cs]: same comment :) Create 2 methods that will check sides :)

**Review by dnnyhlwdplc — APPROVED** [review]

**Pull request merged**